### PR TITLE
Add mix local.hex VERSION

### PR DESCRIPTION
**Pull request by ericmj**

### CODE REVIEW

File: lib/mix/lib/mix/local.ex [hunk 1]
```diff
@@ -199,17 +202,39 @@ defmodule Mix.Local do
     |> Enum.map(&:binary.split(&1, ",", [:global, :trim]))
   end
 
-  defp find_latest_eligible_version(entries) do
-    {:ok, current_version} = Version.parse(System.version())
+  defp find_latest_eligible_version(entries, artifact_version) do
+    elixir_version = Version.parse!(System.version())
```
Comment (whatyouhide) [inline lib/mix/lib/mix/local.ex:206]: Nice!

**Review by whatyouhide — APPROVED** [review]

**Review by josevalim — APPROVED** [review]
Code wise this looks great to me, but I have some food for thought:

1. What is the specified artifact version does not work with Elixir?

2. Today someone can install a specific version with `mix archive.install github hexpm/hex --tag 2.3.4`. Maybe we should document this instead?

**Comment by ericmj** [comment]
1. This error is raised https://github.com/elixir-lang/elixir/pull/11473/files#diff-75787cbe3255513265f665bb859b99a624db2b56bec01e48455cd63577ad8cadR158
2. The idea is to streamline this, make it easer to find how to install a specific version, make it faster, and not require the `git` executable.

**Comment by josevalim** [comment]
Ship it!

**Pull request merged**